src/lazy/expanded/compiler.rs

```diff
@@ -459,7 +484,61 @@ impl TemplateCompiler {
         // The `params` clause of the macro definition is an s-expression enumerating the parameters
         // that the macro accepts. For example: `(flex_uint::x, y*, z?)`.
         let params_clause = Self::expect_sexp("an s-expression defining parameters", &mut values)?;
+        let signature = Self::compile_signature_from_sexp(context, pending_macros, params_clause)?;
```

Comment:
🪧 This diff rendering is unfortunate. I extracted the logic needed to compile a sexp containing parameters (for example: `(foo bar? baz*)`) into a `MacroSignature`, making a new helper method. If you skim ahead to the next large deletion, you'll see that this was mostly just a code block being moved.

The new helper method is now also used when bootstrapping the 1.1 system macro table.